diff --git a/src/grd-session-rdp.c b/src/grd-session-rdp.c
index 7028338..3720a3e 100644
--- a/src/grd-session-rdp.c
+++ b/src/grd-session-rdp.c
@@ -2331,7 +2331,7 @@ grd_session_rdp_remote_desktop_session_started (GrdSession *session)
 
       connector = session_rdp->monitor_config->connectors[0];
       grd_session_record_monitor (session, connector,
-                                  GRD_SCREEN_CAST_CURSOR_MODE_METADATA);
+                                  GRD_SCREEN_CAST_CURSOR_MODE_EMBEDDED);
     }
 }
 
diff --git a/src/grd-session-vnc.c b/src/grd-session-vnc.c
index fbc66a4..9bc4262 100644
--- a/src/grd-session-vnc.c
+++ b/src/grd-session-vnc.c
@@ -728,7 +728,7 @@ close_session_idle (gpointer user_data)
 static void
 grd_session_vnc_remote_desktop_session_started (GrdSession *session)
 {
-  grd_session_record_monitor (session, NULL, GRD_SCREEN_CAST_CURSOR_MODE_METADATA);
+  grd_session_record_monitor (session, NULL, GRD_SCREEN_CAST_CURSOR_MODE_EMBEDDED);
 }
 
 static void
