diff --git a/src/grd-session.c b/src/grd-session.c
index 057bb03..99c1c01 100644
--- a/src/grd-session.c
+++ b/src/grd-session.c
@@ -588,7 +588,7 @@ on_screen_cast_session_proxy_acquired (GObject      *object,
   g_variant_builder_init (&properties_builder, G_VARIANT_TYPE ("a{sv}"));
   g_variant_builder_add (&properties_builder, "{sv}",
                          "cursor-mode",
-                         g_variant_new_uint32 (GRD_SCREEN_CAST_CURSOR_MODE_METADATA));
+                         g_variant_new_uint32 (GRD_SCREEN_CAST_CURSOR_MODE_EMBEDDED));
 
   /* TODO: Support something other than primary monitor */
   grd_dbus_screen_cast_session_call_record_monitor (session_proxy,
